server/stream.go

```diff
@@ -1909,6 +1910,14 @@ func (mset *stream) processMirrorMsgs(mirror *sourceInfo, ready *sync.WaitGroup)
 	}
 }
 
+func (mset *stream) getMirrorHeartbeat() time.Duration {
+	if mset.cfg.Mirror.Heartbeat == 0 {
```

Comment:
@dodomorandi My guess is that the data race is in this method since no read lock is being used (unlike all of the other methods on `mset` when fields are accessed).

Reply:
Actually.. most places the lock is already held except for here: https://github.com/nats-io/nats-server/pull/4095/files#diff-2f4991438bb868a8587303cde9107f83127e88ad70bd19d5c6a31c238a20c299R1874

Reply:
The data race now is fixed, the only thing is that I keep using the initial value of the heartbeat. AFAIK this should not be an issue because I don't expect the value to change across locking blocks. Is my assumption correct or it would be better to use the _current_ value of the heartbeat [here](https://github.com/nats-io/nats-server/pull/4095/files#diff-2f4991438bb868a8587303cde9107f83127e88ad70bd19d5c6a31c238a20c299L2742)?